tests/system/python/packages/test_lab.py

```diff
@@ -0,0 +1,548 @@
+                    assert data[0]["reading"]["event"] == "triggered"
+                    assert data[0]["reading"]["rule"] == "Positive Sine"
+                    break
+            retries -= 1
+
+        if retries == 0:
+            assert "TIMEOUT! positive_sine event not fired." + foglamp_url + "/foglamp/asset/positive_sine?seconds=600"
+
+    def test_negative_sine_notification(self, foglamp_url, remove_data_file, retries, wait_time):
+        remove_data_file("/tmp/out")
+
+        payload = {"name": "Negative Sine", "description": "Negative Sine notification instance", "rule": "Threshold",
+                   "channel": "python35", "notification_type": "retriggered", "enabled": True}
+        post_request(foglamp_url, "/foglamp/notification", payload)
+
+        # Upload Python Script (write_out.py)
+        url = foglamp_url + '/foglamp/category/deliveryNegative%20Sine/script/upload'
+        script_path = 'script=@scripts/write_out.py'
+        upload_script = "curl -sX POST '{}' -F '{}'".format(url, script_path)
+        exit_code = os.system(upload_script)
+        assert exit_code == 0
+
+        payload = {"asset": "sinusoid", "datapoint": "sinusoid", "condition": "<"}
+        put_request(foglamp_url, "/foglamp/category/ruleNegative%20Sine", payload)
+
+        payload = {"enable": "true"}
+        put_request(foglamp_url, "/foglamp/category/deliveryNegative%20Sine", payload)
+
+        for _ in range(retries):
+            if os.path.exists("/tmp/out"):
+                break
+            time.sleep(wait_time*2)
+            print("sleeping, to retry the existence check of '/tmp/out'")
+
+        if not os.path.exists("/tmp/out"):
+            assert False, "TIMEOUT! negative_sine event not fired. No /tmp/out file."
+        else:
+            assert True
+
+    def test_event_toggled_sent_clear(self, foglamp_url, wait_time):
+        print("Add sinusoid")
+        data = {"name": "sin #1", "plugin": "sinusoid", "type": "south", "enabled": True}
+        post_url = "/foglamp/service"
+        data = post_request(foglamp_url, post_url, data)
+        assert data["id"] != "", "ERROR! Failed to add sin #1"
+        assert data["name"] != "", "ERROR! Failed to add sin #1"
+
+        print("Create event instance with threshold and asset; with notification trigger type toggled")
+        data = {"name": "test", "description": "test notification instance", "rule": "Threshold", "channel": "asset",
+                "notification_type": "toggled", "enabled": True}
+        post_url = "/foglamp/notification"
+        post_request(foglamp_url, post_url, data)
+        assert 1
+
+        print("Set rule")
+        data = {"asset": "sinusoid", "datapoint": "sinusoid", "trigger_value": "0.8"}
+        put_url = "/foglamp/category/ruletest"
+        put_request(foglamp_url, put_url, data)
+        assert 1
+
+        print("Set delivery")
+        data = {"asset": "sin0.8", "description": "asset notification", "enable": "true"}
```

Comment:
May be better to create with space like `sin 0.8` to verify assets with spaces works!